db/migrate/20200519073500_remove_visible_from_alchemy_pages.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+class RemoveVisibleFromAlchemyPages < ActiveRecord::Migration[5.2]
+  class LocalPage < ActiveRecord::Base
+    self.table_name = "alchemy_pages"
+  end
+
+  def up
+    if LocalPage.where(visible: [false, nil]).any?
```

Comment:
Existing sites might still have hidden pages (root page or layout pages) that will cause this to fail

Reply:
Also we need a strategy for sites not having the hidden root pages any more and for sites still having them. This makes this migration pretty ugly. We added deprecations, a migration task and upgrade notes into 4.6. I think this extra caution inside a migration (that might run on CIs) is not necessary and could be harming the upgrade.